zerver/lib/webhooks/common.py

```diff
@@ -114,3 +116,32 @@ def validate_extract_webhook_http_header(request: HttpRequest, header: str,
         raise MissingHTTPEventHeader(header)
 
     return extracted_header
+
+class AuthStrategies:
+    SIMPLE_TOKEN = 1
+
+def http_api_callback(bot: UserProfile, uri: str)-> Dict[str, Any]:
+    try:
+        api_callback_details = json.loads(BotConfigData.objects.get(key="api_callback_details").value)
+        auth_strategy = api_callback_details["auth_strategy"]
+    except BotConfigData.DoesNotExist:
+        raise Exception  # TODO: Refine how we notfiy users to configure the auth_details.
+    except KeyError:
+        raise Exception  # TODO
+
+    if int(auth_strategy) == AuthStrategies.SIMPLE_TOKEN:
+        details = {
+            "base_url": api_callback_details["base_url"],
+            "uri": uri,
+            "auth_param": api_callback_details["auth_param"],
+            "auth_token": api_callback_details["auth_token"]
+        }
+        url = "{base_url}{uri}?{auth_param}={auth_token}".format(**details)
+        response = requests.get(url)
+        if not response.status_code == 200:
```

Comment:
This doesn't matter but `if response.status_code != 200` is probably the more Pythonic approach here.

Reply:
Yup, in fact I'll take it a step further and make it `if response.status_code is not 200:`

Reply:
Actually, no that's not good style either. Numerical comparisons should make use of traditional numerical operators for equality tests. :)

Reply:
Ok, I'll stick to `!=` then :)